File: backend/src/main/java/com/smartkrishi/service/weather/WeatherService.java
package com.smartkrishi.service.weather;

import com.smartkrishi.dto.weather.WeatherForecastResponseDTO;
import com.smartkrishi.dto.weather.WeatherResponseDTO;
import java.util.Map;

public interface WeatherService {
    WeatherResponseDTO getCurrentWeather(String city);
    WeatherForecastResponseDTO getForecast(String city);
    WeatherResponseDTO getWeatherByCoordinates(double lat, double lon);
    Map<String, Object> getAirPollution(double lat, double lon);
}

File: frontend/src/services/WeatherService.js
import api from "./api";

const WeatherService = {
  getCurrentWeather: async (city) => {
    const response = await api.get(`/weather/current?city=${encodeURIComponent((city || "").trim())}`, { timeout: 30000 });
    return response.data;
  },

  getForecast: async (city) => {
    const response = await api.get(`/weather/forecast?city=${encodeURIComponent((city || "").trim())}`, { timeout: 30000 });
    return response.data;
  },

  getWeatherByCoordinates: async (lat, lon) => {
    const response = await api.get(`/weather/location`, {
      params: { lat, lon },
      timeout: 30000
    });
    return response.data;
  },

  getAirQuality: async (lat, lon) => {
    const response = await api.get(`/weather/air-quality`, {
      params: { lat, lon },
      timeout: 30000
    });
    return response.data;
  }
};

export default WeatherService;
